### PR TITLE
fix: parseCookie with special chars

### LINKED ISSUE #3401: SSR: split cookie by "=" will cause unexpected issue in some case
### Describe the bug

I am using wagmi with rainbowkit. The cookie of "wagmi.store" contains the `=` string in some fields:

SCR-20240108-mivp

This will cause unterminated json string error when parsing cookie. The cause comes from the following code:

https://github.com/wevm/wagmi/blob/b886c8dcf700344ca8dd43fb71a9e677ae490ce8/packages/core/src/utils/cookie.ts#L29-L34

### Link to Minimal Reproducible Example

_No response_

### Steps To Reproduce

_No response_

### Wagmi Version

2.0.3

### Viem Version

^2

### TypeScript Version

^5

### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/wagmi/issues) for the bug I encountered.

### Anything else?

My suggestion: [split only the first `=`](https://stackoverflow.com/a/50452780/7422256)

I'm not sure if there is still some other edge case though.

I'm happy to make a PR if you think this is ok.

### PR DESCRIPTION
**Pull request by Songkeys**
## Description

fix #3401 

## Additional Information

Before submitting this issue, please make sure you do the following.

- [x] Read the [contributing guide](https://wagmi.sh/dev/contributing)
- [x] Added documentation related to the changes made.
- [x] Added or updated tests (and snapshots) related to the changes made.

### CODE REVIEW

File: .changeset/tender-keys-travel.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@wagmi/core": patch
+---
+
+fix parseCookie in ssr when the cookie value field has special chars "="
```
Comment (tmm) [inline .changeset/tender-keys-travel.md:5]:
```suggestion
Fixed SSR cookie support for cookies that have special characters, e.g. `=`.
```

File: .changeset/tender-keys-travel.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@wagmi/core": patch
```
Comment (tmm) [inline .changeset/tender-keys-travel.md:2]: Adding Wagmi so it pulls in this fix.

```suggestion
"@wagmi/core": patch
"wagmi": patch
```

File: packages/core/src/utils/cookie.test.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { expect, test } from 'vitest'
+
+import { parseCookie } from './cookie.js'
+
+test('parseCookie', () => {
```
Comment (tmm) [inline packages/core/src/utils/cookie.test.ts:5]: No need to update now, but [`test.each`](https://vitest.dev/api/#test-each) could be good for something like this in the future.

File: packages/core/src/utils/cookie.ts [hunk 4]
```diff
@@ -27,8 +27,9 @@ export function cookieToInitialState(config: Config, cookie?: string | null) {
...
 
 export function parseCookie(cookie: string, key: string) {
-  return cookie
-    .split('; ')
-    .find((x) => x.startsWith(key))
-    ?.split('=')[1]
+  const keyValue = cookie.split('; ').find((x) => x.startsWith(`${key}=`))
+  if (!keyValue) {
+    return undefined
+  }
```
Comment (tmm) [inline packages/core/src/utils/cookie.ts:33]:
```suggestion
  if (!keyValue) return undefined
```

File: .changeset/tender-keys-travel.md (hunk 2 above)
Comment (Songkeys) [reply .changeset/tender-keys-travel.md:2]: Do I need to add `@wagmi/connectors` as well?

SCR-20240109-jlcf

File: packages/core/src/utils/cookie.test.ts (hunk 3 above)
Comment (Songkeys) [reply packages/core/src/utils/cookie.test.ts:5]: Did you meant I should remove this test file for this PR?

**Pull request merged**